pkg/controller/addroletoinstanceprofile/addroletoinstanceprofile_controller.go

```diff
@@ -129,6 +129,7 @@ func (r *ReconcileAddRoleToInstanceProfile) Reconcile(request reconcile.Request)
 	// if absent then create
 	iamRoleAddedToInstanceProfile, ok := instance.ObjectMeta.Annotations[`iamRoleAddedToInstanceProfile`]
 	if !ok {
+		instance.ObjectMeta.Annotations = make(map[string]string) // This keeps the controller from crashing when annotation later on
```

Comment:
Can you move this line to just before you start putting things into the map?

Reply:
Done. Did the same with all of the other controllers.